File: design-patterns/go/visitor/hosting_cost/hosting_calculator_visitor.go
// hosting_calculator_visitor.go

package main

type HostingCalculatorVisitor interface {
	ComputeVisit(computeService *ComputeService) float64
	ContainerVisit(containerService *ContainerService) float64
	DatabaseVisit(databaseService *DatabaseService) float64
	FileVisit(fileStorageService *FileStorageService) float64
	ServerlessVisit(serverlessService *ServerlessService) float64
}

File: design-patterns/php/visitor/hostingcost/HostingCalculatorVisitor.php
<?php
// HostingCalculatorVisitor.php

namespace BigBoxCode\DesignPattern\Visitor\HostingCost;

interface HostingCalculatorVisitor {
    public function visitComputeService(ComputeService $computeService): float;
    public function visitDatabaseService(DatabaseService $databaseService): float;
    public function visitFileStorageService(FileStorageService $fileStorageService): float;
    public function visitServerlessService(ServerlessService $serverlessService): float;
    public function visitContainerService(ContainerService $containerService): float;
}